Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM ubuntu:16.04
```

Comment:
If docker works well enough it's possible that we might be able to look into it again for use instead of the NUbots vm in vagrant. Docker has a few advantages over it.

Reply:
Yeah @Bidski has mentioned this. After this is merged I will be running some tests to see if it works well on Windows for general development. That will probably be a different PR though.

Reply:
Agreed that it should be a separate PR. However, if you do that it also means that we will replace puppet with docker.

Reply:
In progress in #255.